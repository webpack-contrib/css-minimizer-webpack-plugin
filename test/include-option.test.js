import CssMinimizerPlugin from "../src/index";

import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readAssets,
} from "./helpers";

describe("include option", () => {
  let compiler;

  beforeEach(() => {
    compiler = getCompiler({
      entry: {
        included1: `${__dirname}/fixtures/included1.js`,
        included2: `${__dirname}/fixtures/included2.js`,
        entry: `${__dirname}/fixtures/entry.js`,
      },
    });
  });

  it("should match snapshot for a single RegExp value included1", async () => {
    new CssMinimizerPlugin({
      include: /included1/i,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for a single String value included1", async () => {
    new CssMinimizerPlugin({
      include: "included1",
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for multiple RegExp values included1 and included2", async () => {
    new CssMinimizerPlugin({
      include: [/included1/i, /included2/i],
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for multiple String values included1 and included2", async () => {
    new CssMinimizerPlugin({
      include: ["included1", "included2"],
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
