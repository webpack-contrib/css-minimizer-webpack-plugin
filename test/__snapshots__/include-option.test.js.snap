// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`include option should match snapshot for a single RegExp value included1: assets 1`] = `
{
  "entry.css": "body {
  color: red;
}
a {
  color: blue;
}
",
  "included1.css": "a{text-align:center}",
  "included2.css": "body {
  color: red;
}
body {
  color: red;
  font-size: 20px;
}
a {
  color: blue;
}

",
}
`;

exports[`include option should match snapshot for a single RegExp value included1: errors 1`] = `[]`;

exports[`include option should match snapshot for a single RegExp value included1: warnings 1`] = `[]`;

exports[`include option should match snapshot for a single String value included1: assets 1`] = `
{
  "entry.css": "body {
  color: red;
}
a {
  color: blue;
}
",
  "included1.css": "a{text-align:center}",
  "included2.css": "body {
  color: red;
}
body {
  color: red;
  font-size: 20px;
}
a {
  color: blue;
}

",
}
`;

exports[`include option should match snapshot for a single String value included1: errors 1`] = `[]`;

exports[`include option should match snapshot for a single String value included1: warnings 1`] = `[]`;

exports[`include option should match snapshot for multiple RegExp values included1 and included2: assets 1`] = `
{
  "entry.css": "body {
  color: red;
}
a {
  color: blue;
}
",
  "included1.css": "a{text-align:center}",
  "included2.css": "body{color:red;font-size:20px}a{color:blue}",
}
`;

exports[`include option should match snapshot for multiple RegExp values included1 and included2: errors 1`] = `[]`;

exports[`include option should match snapshot for multiple RegExp values included1 and included2: warnings 1`] = `[]`;

exports[`include option should match snapshot for multiple String values included1 and included2: assets 1`] = `
{
  "entry.css": "body {
  color: red;
}
a {
  color: blue;
}
",
  "included1.css": "a{text-align:center}",
  "included2.css": "body{color:red;font-size:20px}a{color:blue}",
}
`;

exports[`include option should match snapshot for multiple String values included1 and included2: errors 1`] = `[]`;

exports[`include option should match snapshot for multiple String values included1 and included2: warnings 1`] = `[]`;
